File: Board/src/main/java/com/example/board/global/error/ErrorResponse.java
package com.example.board.global.error;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE) //클래스에 존재하는 모든 필드에 대한 생성자를 자동으로 생성
@Builder //생성자를 호출할 수 있게 빌더 클래스를 생성
public class ErrorResponse {
    private final int status;
    private final String message;
}

File: Kopring/Dise_V2/src/main/kotlin/com/example/dise_v2/global/error/ErrorResponse.kt
package com.example.dise_v2.global.error

class ErrorResponse(
    val status: Int,
    val message: String
) {
    companion object {
        fun of(e:  CustomException) : ErrorResponse {
            return ErrorResponse(
                status = e.status,
                message = e.message
            )
        }
    }
}